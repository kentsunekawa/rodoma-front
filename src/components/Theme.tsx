import React from 'react';
import { ThemeProvider } from 'styled-components';
import AppTheme from 'components/style/AppTheme';
import { useSelector } from 'react-redux';
import { modeSelector } from 'state/modules/app';
import GlobalStyle from './style/GlobalStyle';

interface Props {
  children: React.ReactNode;
}

const Component: React.FC<Props> = (props: Props) => {
  const mode = useSelector(modeSelector);

  return (
    <ThemeProvider theme={AppTheme[mode]}>
      <GlobalStyle />
      {props.children}
    </ThemeProvider>
  );
};

export default Component;
