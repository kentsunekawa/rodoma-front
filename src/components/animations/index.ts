import gsap from "gsap";

export const doorHide = (root: HTMLDivElement, logo: HTMLDivElement) => {
  return new Promise((resolve) => {
    gsap.to(logo, {
      y: "-200px",
      duration: 1,
      ease: "Power3.easeOut",
    });
    gsap.to(root, {
      opacity: 0,
      duration: 0.5,
      ease: "linear",
      onComplete: () => {
        gsap.set(root, {
          display: "none",
        });
        resolve(null);
      },
    });
  });
};

export const toggleLoading = (root: HTMLDivElement, dir: boolean = true) => {
  if (dir) {
    return new Promise((resolve) => {
      gsap.killTweensOf(root);
      gsap.set(root, {
        display: "block",
        opacity: "0",
      });
      gsap.to(root, {
        opacity: 1,
        duration: 0.2,
        ease: "linear",
        onComplete: () => {
          resolve(null);
        },
      });
    });
  } else {
    return new Promise((resolve) => {
      gsap.killTweensOf(root);
      gsap.to(root, {
        opacity: 0,
        duration: 0.2,
        ease: "linear",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    });
  }
};

export const toggleMenu = (
  root: HTMLDivElement,
  panel: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  if (dir) {
    gsap.set(root, {
      display: "block",
    });
    gsap.set(overlay, {
      opacity: 0,
    });
    gsap.set(panel, {
      x: "100%",
    });
    gsap.to(overlay, {
      opacity: 1,
      duration: 0.3,
      ease: "Power1.easeInOut",
    });
    gsap.to(panel, {
      x: "0",
      duration: 0.3,
    });
  } else {
    gsap.to(overlay, {
      opacity: 0,
      duration: 0.3,
    });
    gsap.to(panel, {
      x: "100%",
      duration: 0.3,
      ease: "Power1.easeInOut",
      onComplete: () => {
        root.style.display = "none";
      },
    });
    gsap.set(root, {
      display: "none",
      delay: 0.3,
    });
  }
};

export const toggleAccordion = (
  child: HTMLDivElement,
  inner: HTMLDivElement,
  dir: boolean = true
) => {
  const ease = "Power2.easeInOut";
  const duration = 0.3;

  if (dir) {
    gsap.set(child, {
      height: 0,
    });
    gsap.to(child, {
      height: inner.clientHeight,
      duration,
      ease,
    });
  } else {
    gsap.to(child, {
      height: 0,
      duration,
      ease,
    });
  }
};

export const toggleSearchPanel = (
  bg: HTMLDivElement,
  inner: HTMLDivElement,
  dir: boolean = true
) => {
  const ease = "Power2.easeOut";
  const duration = 0.3;

  if (dir) {
    gsap.set(bg, {
      display: "block",
      opacity: 0,
    });
    gsap.set(inner, {
      opacity: 0,
      y: "50px",
    });
    gsap.to(bg, {
      opacity: 1,
      duration,
      ease: "Power0.easeNone",
    });
    gsap.to(inner, {
      y: 0,
      opacity: 1,
      duration,
      ease,
      delay: 0.1,
    });
  } else {
    gsap.to(inner, {
      y: "50px",
      opacity: 0,
      duration,
      ease,
    });
    gsap.to(bg, {
      opacity: 0,
      duration,
      ease: "Power0.easeNone",
      // delay: .2,
      onComplete: () => {
        bg.style.display = "none";
      },
    });
  }
};

export const toggleModal = (
  root: HTMLDivElement,
  mask: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  const duration = 0.3;
  const ease = "Power2.easeOut";

  return new Promise((resolve) => {
    if (dir) {
      gsap.set(root, {
        display: "block",
      });
      gsap.set(overlay, {
        opacity: 0,
      });
      gsap.set(mask, {
        y: "50px",
        opacity: 0,
      });
      gsap.to(overlay, {
        opacity: 1,
        duration,
        ease: "Power0.easeNone",
      });
      gsap.to(mask, {
        y: 0,
        opacity: 1,
        duration,
        ease,
        onComplete() {
          resolve(null);
        },
      });
    } else {
      gsap.to(overlay, {
        opacity: 0,
        duration,
        ease: "Power0.easeNone",
      });
      gsap.to(mask, {
        y: "50px",
        opacity: 0,
        duration,
        ease,
        onComplete() {
          resolve(null);
        },
      });
    }
  });
};

export const toggleMessage = (
  root: HTMLDivElement,
  counter: HTMLSpanElement,
  isShow: boolean = true,
  count?: number
) => {
  return new Promise((resolve) => {
    if (isShow) {
      gsap.killTweensOf(counter);
      gsap.set(root, {
        display: "block",
        opacity: 0,
      });
      gsap.set(counter, {
        width: 0,
      });
      gsap.to(counter, {
        width: "100%",
        duration: count,
        ease: "linear",
      });
      gsap.to(root, {
        opacity: 1,
        duration: 0.3,
        ease: "Power0.easeNone",
        onComplete: () => {
          resolve(null);
        },
      });
    } else {
      gsap.to(root, {
        opacity: 0,
        duration: 0.3,
        ease: "Power0.easeNone",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    }
  });
};

export const toggleCard = (
  root: HTMLDivElement,
  panel: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  const duration = 0.4;

  return new Promise((resolve) => {
    if (dir) {
      gsap.set(root, {
        display: "block",
      });
      gsap.set(overlay, {
        opacity: 0,
      });
      gsap.set(panel, {
        bottom: "-100%",
      });
      gsap.to(overlay, {
        opacity: 1,
        duration,
        ease: "linear",
      });
      gsap.to(panel, {
        bottom: 0,
        duration,
        ease: "Power3.easeOut",
        onComplete: () => {
          resolve(null);
        },
      });
    } else {
      gsap.to(panel, {
        bottom: "-100%",
        duration,
        ease: "Power3.easeIn",
      });
      gsap.to(overlay, {
        opacity: 0,
        duration,
        ease: "linear",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    }
  });
};
