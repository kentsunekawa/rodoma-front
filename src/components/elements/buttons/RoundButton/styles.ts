import { css } from 'styled-components';
import { Gradient, ButtonTapAction, WithShadow } from 'components/style/Mixins';

export const base = css`
  position: relative;
  display: flex;
  align-items: center;
  align-content: center;
  justify-content: center;
  padding: 0 1em;
  height: 40px;
  border-radius: 20px;
  font-size: 1.5rem;
  color: ${({ theme }) => theme.colors.primary};
  background: #fff;
  .icon {
    position: absolute;
    top: 50%;
    right: 0.3em;
    transform: translateY(-50%);
    display: block;
    font-size: 150%;
  }
  ${WithShadow}
  ${ButtonTapAction}
  &.-disabled {
    opacity: 0.5;
  }
`;

export const s = css`
  height: 30px;
  border-radius: 15px;
  font-size: 1.2rem;
`;

export const m = css`
  height: 40px;
  border-radius: 20px;
  font-size: 1.5rem;
`;

export const l = css`
  height: 60px;
  border-radius: 30px;
  padding: 0 2em;
  font-size: 2rem;
`;

export const gray_light = css`
  background: ${({ theme }) => theme.colors.gray_light};
  color: #fff;
`;

export const gray_midium = css`
  background: ${({ theme }) => theme.colors.gray_midium};
  color: #fff;
`;

export const gray_dark = css`
  background: ${({ theme }) => theme.colors.gray_dark};
  color: #fff;
`;

export const gradient = css`
  ${Gradient}
  color: #fff;
`;

export const nega = css`
  color: #fff;
`;

export const withIcon = css`
  padding: 0 2.5em;
`;
