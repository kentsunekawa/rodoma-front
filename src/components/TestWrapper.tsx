import React from 'react';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from 'state/store';
import Theme from 'components/Theme';

interface Props {
  children?: React.ReactNode;
}

const Component: React.FC<Props> = (props: Props) => (
  <Provider store={store}>
    <Theme>
      <Router>{props.children}</Router>
    </Theme>
  </Provider>
);

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  render(ui, { wrapper: Component, ...options });

export * from '@testing-library/react';

export { customRender as render };
