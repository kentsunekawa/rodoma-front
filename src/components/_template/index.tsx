import React from 'react';
import Styled from 'styled-components';
import * as styles from './styles';

// component root class name
const CLASSNAME = '';

// declare types
type StyleType = 'primary';

interface ComponentProps {
  className?: string;
  types?: StyleType[];
}

interface Props extends ComponentProps {}

// dom component
const Component: React.FC<Props> = (props) => (
  <div className={`${CLASSNAME} ${props.className}`}></div>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
  // extended styles
  ${({ types }) => types && types.map((type) => styles[type])}}
`;

// container component
const Container: React.FC<ComponentProps> = (componentProps) => {
  const props = {};

  return <StyeldComponent {...componentProps} {...props}></StyeldComponent>;
};
export default Container;
