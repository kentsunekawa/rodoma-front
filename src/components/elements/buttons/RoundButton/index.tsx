import React from 'react';
import { useHistory } from 'react-router-dom';
import Styled from 'styled-components';
import * as styles from './styles';

// component root class name
const CLASSNAME = 'RoundButton';

// declare types
type StyleType =
  | 's'
  | 'm'
  | 'l'
  | 'gradient'
  | 'gray_dark'
  | 'gray_midium'
  | 'gray_light'
  | 'nega'
  | 'withIcon';

export interface ComponentProps {
  disabled?: boolean;
  className?: string;
  types?: StyleType[];
  icon?: React.ReactNode;
  link?: string;
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

interface Props extends ComponentProps {
  children: React.ReactNode;
  clickFunc: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

// dom component
export const Component: React.FC<Props> = (props: Props) => (
  <>
    {!props.link && !props.onClick ? (
      <div className={`${CLASSNAME} ${props.className}${props.disabled ? ' -disabled' : ''}`}>
        {props.children}
        {props.icon && <div className="icon">{props.icon}</div>}
      </div>
    ) : (
      <button
        className={`${CLASSNAME} ${props.className}${props.disabled ? ' -disabled' : ''}`}
        onClick={props.clickFunc}
      >
        {props.children}
        {props.icon && <div className="icon">{props.icon}</div>}
      </button>
    )}
  </>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
  // extended styles
  ${({ types }) => types && types.map((type) => styles[type])}}
`;

// container component
const Container: React.FC<ComponentProps> = (componentProps) => {
  const history = useHistory();

  const { disabled, link, onClick } = componentProps;

  const clickFunc = (e: React.MouseEvent<HTMLButtonElement>) => {
    if (disabled !== true) {
      if (onClick) onClick(e);
      if (link) history.push(link);
    }
  };

  const props = { clickFunc };

  return (
    <StyeldComponent {...componentProps} {...props}>
      {componentProps.children}
    </StyeldComponent>
  );
};

export default Container;
