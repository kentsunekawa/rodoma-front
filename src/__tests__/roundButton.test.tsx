import React from 'react';
// import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'components/TestWrapper';
import RoundButton, { Component } from 'components/elements/buttons/RoundButton';

let container: HTMLDivElement | null = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(cleanup);

describe('RoundButton', () => {
  test('RoundButton のスナップショットテスト', () => {
    const props = {
      clickFunc: jest.fn,
    };
    const tree = renderer.create(<Component {...props}>button</Component>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ボタンテキストが正しく表示されているか', () => {
    const tree = render(<RoundButton>button</RoundButton>);

    expect(tree.container.textContent).toBe('button');
  });

  test('click イベントの確認', () => {
    const props = {
      onClick: jest.fn,
    };
    const spyClick = jest.spyOn(props, 'onClick');

    const tree = render(<RoundButton {...props}>button</RoundButton>);

    const button = tree.container.querySelector('button');
    if (button !== null) {
      fireEvent.click(button);
    }
    expect(spyClick).toHaveBeenCalled();
  });
});
