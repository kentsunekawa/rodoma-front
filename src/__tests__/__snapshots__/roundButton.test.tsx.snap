// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton RoundButton のスナップショットテスト 1`] = `
<div
  className="RoundButton undefined"
>
  button
</div>
`;

exports[`RoundButton snpashot test 1`] = `
<div
  className="RoundButton undefined"
>
  button
</div>
`;
